haskell/experimental/private/module.bzl

```diff
@@ -183,6 +184,20 @@ def _build_haskell_module(
     # Construct compiler arguments
 
     args = ctx.actions.args()
+
+    main_function = getattr(ctx.attr, "main_function", None)
+
+    if main_function:
+        main_file = getattr(ctx.attr, "main_file", None)
+        main_function_module = infer_main_module(main_function)
+        if moduleAttr.module_name:
+            guess_module_name = moduleAttr.module_name
+        else:
+            guess_module_name = paths.split_extension(get_module_path_from_target(module))[0].replace("/", ".")
```

Comment:
Doesn't `get_module_path_from_target` already do the `.hs` removal internally? This seems redundant here.

Reply:
Fixed it